src/Gateways/AbstractGateway.php

```diff
@@ -769,6 +769,75 @@ protected function add_hooks() {
 				'woocommerce_available_payment_gateways'
 			) );
 		}
+
+		add_action( 'woocommerce_api_globalpayments_test_refund', array(
+			$this,
+			'test_refund_programmatically'
+		) );
+	}
+
+	/**
+	 * Test endpoint for testing a programmatically refund instead of using admin
+	 *
+	 * URL ENDPOINT www.domain.com/?wc-api=globalpayments_test_refund&order-id=136&reason=something&amount=1
+	 *
+	 * @param int $order-id // valid order-id from a refundable order.
+	 *
+	 * @param string $reason // reason of refund.
+	 *
+	 * @param int $amount // amount of refund , only integer for testing.
+	 *
+	 * @return array
+	 */
+	public function test_refund_programmatically () {
```

Comment:
@javiercrac if we want to use this endpoint for QA, it shouldn't do any validation data. It should just send the data and print the response.

These are things that the process_refund method should be able to deal with and respond accordingly.

e.g. null amount, invalid amount format "0,99", etc.

In case of any error the endpoint response should replicate the response received after clicking "Refund" in the Admin Dashboard.

Reply:
I removed those validations in the endpoint.